fastapi/dependencies/utils.py

```diff
@@ -535,25 +535,29 @@ async def solve_dependencies(
         if sub_errors:
             errors.extend(sub_errors)
             continue
-        if sub_dependant.use_cache and sub_dependant.cache_key in dependency_cache:
-            solved = dependency_cache[sub_dependant.cache_key]
-        elif is_gen_callable(call) or is_async_gen_callable(call):
-            stack = request.scope.get("fastapi_astack")
-            if stack is None:
-                raise RuntimeError(
-                    async_contextmanager_dependencies_error
-                )  # pragma: no cover
-            solved = await solve_generator(
-                call=call, stack=stack, sub_values=sub_values
-            )
-        elif is_coroutine_callable(call):
-            solved = await call(**sub_values)
-        else:
-            solved = await run_in_threadpool(call, **sub_values)
-        if sub_dependant.name is not None:
-            values[sub_dependant.name] = solved
-        if sub_dependant.cache_key not in dependency_cache:
-            dependency_cache[sub_dependant.cache_key] = solved
+
+        try:
+            if sub_dependant.use_cache and sub_dependant.cache_key in dependency_cache:
+                solved = dependency_cache[sub_dependant.cache_key]
+            elif is_gen_callable(call) or is_async_gen_callable(call):
+                stack = request.scope.get("fastapi_astack")
+                if stack is None:
+                    raise RuntimeError(
+                        async_contextmanager_dependencies_error
+                    )  # pragma: no cover
+                solved = await solve_generator(
+                    call=call, stack=stack, sub_values=sub_values
+                )
+            elif is_coroutine_callable(call):
+                solved = await call(**sub_values)
+            else:
+                solved = await run_in_threadpool(call, **sub_values)
+            if sub_dependant.name is not None:
+                values[sub_dependant.name] = solved
+            if sub_dependant.cache_key not in dependency_cache:
+                dependency_cache[sub_dependant.cache_key] = solved
+        except Exception as e:
```

Comment:
Is catching all `Exception` errors overly greedy? Perhaps we should only handle `ValueError` and `TypeError`, as those seem to be the two bases for all of Pydantic's validation errors (I may be wrong on that, I'm not that well versed in Pydantic):

https://github.com/samuelcolvin/pydantic/blob/c256dccbb383a7fd462f62fcb5d55558eb3cb108/pydantic/errors.py#L128-L133
```suggestion
        except (ValueError, TypeError) as e:
```